uaclient/tests/test_cli.py

```diff
@@ -828,208 +829,54 @@ def test_status_human_readable_warning(
-    @mock.patch("uaclient.log.get_user_log_file")
-    def test_console_log_configured_if_not_present(
+    def test_correct_handlers_added_to_logger(
         self,
-        m_get_user,
-        m_we_are_currently_root,
-        level,
-        capsys,
         logging_sandbox,
         FakeConfig,
-        tmpdir,
     ):
-        m_get_user.return_value = tmpdir.join("user.log").strpath
-        with mock.patch(
-            "uaclient.cli.config.UAConfig", return_value=FakeConfig()
-        ):
-            setup_logging(level, logging.INFO)
-        logging.log(level, "after setup")
-        logging.log(level - 1, "not present")
-
-        _, err = capsys.readouterr()
-        assert "after setup" in err
-        assert "not present" not in err
+        console_level = logging.INFO
+        log_level = logging.DEBUG
+        logger = logging.getLogger("logger_a")
 
-    @mock.patch("uaclient.cli.util.we_are_currently_root", return_value=False)
-    @mock.patch("uaclient.log.get_user_log_file")
-    def test_console_log_configured_if_already_present(
-        self,
-        m_get_user,
-        m_we_are_currently_root,
-        capsys,
-        logging_sandbox,
-        FakeConfig,
-        tmpdir,
-    ):
-        m_get_user.return_value = tmpdir.join("user.log").strpath
-        logging.getLogger().addHandler(logging.StreamHandler(sys.stderr))
+        handler = logging.StreamHandler(sys.stderr)
+        handler.setLevel(logging.ERROR)
+        handler.set_name("ua-test-console")
+        logger.addHandler(handler)
 
         with mock.patch(
             "uaclient.cli.config.UAConfig", return_value=FakeConfig()
         ):
-            logging.error("before setup")
-            setup_logging(logging.INFO, logging.INFO)
-            logging.error("after setup")
-
-        # 'before setup' will be in stderr, so check that setup_logging
-        # configures the format
-        _, err = capsys.readouterr()
-        assert "ERROR: before setup" not in err
-        assert "ERROR: after setup" in err
+            setup_logging(console_level, log_level, logger=logger)
+        assert len(logger.handlers) == 2
+        for idx, handler in enumerate(logger.handlers):
+            if idx == 0:
+                assert handler.name == "ua-console"
+                assert handler.level == console_level
+            elif idx == 1:
+                assert handler.name == "ua-file"
+                assert handler.level == log_level
+
+    @mock.patch("pathlib.Path.touch")
+    def test_log_file_created_if_not_present(self, m_path_touch, tmpdir):
+        logger = logging.getLogger("logger_b")
+        log_file = tmpdir.join("log_file").strpath
+        setup_logging(
+            logging.DEBUG,
+            logging.INFO,
+            log_file=log_file,
+            logger=logger,
+        )
+        assert m_path_touch.call_count == 1
+        assert m_path_touch.call_args_list == [mock.call(mode=0o640)]
```

Comment:
asserting the call_args_list is what it is means that it was only called once, so no need for the call_count assertion
```suggestion
        assert m_path_touch.call_args_list == [mock.call(mode=0o640)]
```